simulator/test_simulator.py

```diff
@@ -0,0 +1,270 @@
+        self.assertEqualFloat(state.acceleration, target)
+
+        delta_t, step_n = simulator.run(1.0)
+
+        delta_v = target * delta_t
+        delta_x = eval_path_length(state.velocity, target, step_n, time_step)
+
+        self.assertEqualFloat(simulator.get_target(), target)
+
+        state = simulator.get_state()
+        self.assertEqualFloat(state.acceleration, target)
+        self.assertEqualFloat(state.velocity, before.velocity + delta_v)
+        self.assertEqualFloat(state.position, before.position + delta_x)
+
+    def test_complex_acc(self):
+        '''
+        A sequential series of accelerations.
+        1. acceleration
+        2. uniform velocity
+        3. full braking
+        '''
+
+        time_step = self.physical_params.time_step
+        simulator = self.simulator
+
+        before = simulator.get_state()
+
+        # acceleration
+        target = 0.2
+        simulator.set_target(target)
+
+        delta_t, step_n = simulator.run(0.3)
+        delta_v = target * delta_t
+        delta_x = eval_path_length(before.velocity, target, step_n, time_step)
+
+        state = simulator.get_state()
+        self.assertEqualFloat(state.acceleration, target)
+        self.assertEqualFloat(state.velocity, before.velocity + delta_v)
+        self.assertEqualFloat(state.position, before.position + delta_x)
+
+        before = state
+
+        # uniform
+        target = 0.0
+        simulator.set_target(target)
+        delta_t, step_n = simulator.run(0.3)
+        delta_v = target * delta_t
+        delta_x = eval_path_length(before.velocity, target, step_n, time_step)
+
+        state = simulator.get_state()
+        self.assertEqualFloat(state.acceleration, target)
+        self.assertEqualFloat(state.velocity, before.velocity + delta_v)
+        self.assertEqualFloat(state.position, before.position + delta_x)
+
+        before = state
+
+        # breaking
+        target = 0.6
+        simulator.set_target(target)
+        delta_t, step_n = simulator.run(0.1)
+        delta_v = target * delta_t
+        delta_x = eval_path_length(before.velocity, target, step_n, time_step)
+
+        state = simulator.get_state()
+        self.assertEqualFloat(state.acceleration, target)
+        self.assertEqualFloat(state.velocity, before.velocity + delta_v)
+        self.assertEqualFloat(state.position, before.position + delta_x)
+
+class AccelerationClampTest(unittest.TestCase):
+    def test_clamp(self):
+        '''
+        Simulator limits acceleration with clipping.
+        '''
+
+        simulator = CartPoleSimulator()
+        simulator.reset_physical_params(PhysicalParams())
+
+        config = Config.default()
+        config.max_acceleration = 1.0
+        config.clamp_acceleration = False
+        simulator.reset(config)
+
+        simulator.set_target(config.max_acceleration + 1)
+        self.assertAlmostEqual(simulator.get_target(), config.max_acceleration)
```

Comment:
as we discussed, this should either clamp or raise an error. but here, as clamping is disabled, i think we should expect an exception

Reply:
fix